soh/soh/Enhancements/randomizer/3drando/location_access/locacc_gerudo_training_grounds.cpp

```diff
@@ -95,7 +95,7 @@ void AreaTable_Init_GerudoTrainingGrounds() {
 
   areaTable[GERUDO_TRAINING_GROUNDS_HEAVY_BLOCK_ROOM] = Area("Gerudo Training Grounds Heavy Block Room", "Gerudo Training Grounds", GERUDO_TRAINING_GROUNDS, NO_DAY_NIGHT_CYCLE, {}, {
                   //Locations
-                  LocationAccess(GERUDO_TRAINING_GROUNDS_BEFORE_HEAVY_BLOCK_CHEST, {[]{return true;}}),
+                  LocationAccess(GERUDO_TRAINING_GROUNDS_BEFORE_HEAVY_BLOCK_CHEST, {[]{return (CanUseSword || CanUse(MEGATON_HAMMER));}}),
```

Comment:
Need to have items here to defeat the wolfos etc. Defaulted to sword + hammer as there are 4 enemies and these are the only options which dont use up consumables

Reply:
Aren't there other places in the logic that incorporate consumables into the equation? Isn't that what `CanDamage` is for?

Reply:
Yeah they do but it kinda mixes between whether they do or not, typically I've found consumables are taken into account when there's a fewer amount of enemies. I am planning on adding a trick for like advanced combat where any possible way to beat an enemy is taken into account since there's several of these occassions. For this specific change it won't end up changing anything since the glitchless logic requires you to have Sword/Hammer from earlier (which is why it just said 'true' before) so this is mostly for the benefit of glitched logic

Reply:
Oh, nevermind, my mind is playing tricks on me. Could have sworn there was a `CanDamage` in there. maybe it's named something else

Reply:
Typically what was previously used was a combination of (IsAdult || CanChildDamage) or (IsAdult || CanChildAttack) which is what I'm trying to move against. As I find out what combinations of items end up being more common I'm gonna make more CanXXXXX things

Reply:
If we're finally getting away from the assumption that adult has the MS, then you might want to *make* the `CanDamage` to roll all that into.

Reply:
Yea thats the plan 👍🏼

Reply:
Out of curiousity, have you looked at how 3DS handles these rooms right now, considering they have master sword shuffle implemented?

Reply:
Had looked for other rooms but assumed this one they'd have kept as just 'true'. looks like they have CanJumpslash so uses sword + sticks but doesnt account for hammer (or bow/other consumables) .  Seems weird to me to allow some consumables and not others so would personally prefer sticking to a general 'no consumables' policy for rooms with many enemies/strong enemies, and then later move to adding a trick option to allow with consumables